src/Workspaces/Core/Portable/Telemetry/TelemetryLogging.cs

```diff
@@ -134,13 +123,15 @@ public static void Flush()
         Flushed?.Invoke(null, EventArgs.Empty);
     }
 
-    private static ValueTask PostCollectedTelemetryAsync(CancellationToken cancellationToken)
+    private static async Task PostCollectedTelemetryAsync(CancellationToken cancellationToken)
     {
-        Flush();
+        await Task.Delay(TimeSpan.FromMinutes(30), cancellationToken).ConfigureAwait(false);
 
-        // Ensure PostCollectedTelemetryAsync will get fired again after the collection period.
-        s_postTelemetryQueue?.AddWork();
+        Flush();
 
-        return ValueTaskFactory.CompletedTask;
+        // Create a fire and forget task to handle the next collection. This doesn't use IAsynchronousOperationListener
+        // to track this work as no-one needs to ensure this is sent, and the create a new item of work
+        // upon previous completion doesn't fit well in that model.
+        _ = PostCollectedTelemetryAsync(CancellationToken.None);
```

Comment:
consider wrapping with reportnonfatal (so we'll have non fatals if something goes wrong)